python/ray/train/tests/test_huggingface_trainer.py

```diff
@@ -104,38 +106,6 @@ def test_e2e(ray_start_4_cpus, save_strategy):
     assert predictions.count() == 3
 
 
-def test_reporting():
-    reports = []
-
-    def _fake_report(**kwargs):
-        reports.append(kwargs)
-
-    with patch("ray.air.session.report", _fake_report):
-        state = TrainerState()
-        report_callback = TrainReportCallback()
-        report_callback.on_epoch_begin(None, state, None)
-        state.epoch = 0.5
-        report_callback.on_log(None, state, None, logs={"log1": 1})
-        state.epoch = 1
-        report_callback.on_log(None, state, None, logs={"log2": 1})
-        report_callback.on_epoch_end(None, state, None)
-        report_callback.on_epoch_begin(None, state, None)
-        state.epoch = 1.5
-        report_callback.on_log(None, state, None, logs={"log1": 1})
-        state.epoch = 2
-        report_callback.on_log(None, state, None, logs={"log2": 1})
-        report_callback.on_epoch_end(None, state, None)
-        report_callback.on_train_end(None, state, None)
-
-    assert len(reports) == 2
-    assert "log1" in reports[0]["metrics"]
-    assert "log2" in reports[0]["metrics"]
-    assert reports[0]["metrics"]["epoch"] == 1
-    assert "log1" in reports[1]["metrics"]
-    assert "log2" in reports[1]["metrics"]
-    assert reports[1]["metrics"]["epoch"] == 2
```

Comment:
we don't need this test anymore?

Reply:
Yeah. This test was only applicable to epoch testing, and it was only necessary due to poor validation. Now that we are validating epoch vs steps correctly, it can be safely removed. The e2e tests we have cover logging well enough.